feature/account/common/src/main/kotlin/app/k9mail/feature/account/common/domain/entity/Folder.kt

```diff
@@ -0,0 +1,53 @@
+package app.k9mail.feature.account.common.domain.entity
+
+import com.fsck.k9.mail.folders.RemoteFolder
+
+sealed interface Folder {
+    data class None(
+        val isAutomatic: Boolean = false,
+    ) : Folder
+
+    data class Regular(
+        val remoteFolder: RemoteFolder,
+    ) : Folder
+}
+
+sealed interface SpecialFolder : Folder {
+    val remoteFolder: RemoteFolder
+    val isAutomatic: Boolean get() = false
+
+    data class Inbox(
+        override val remoteFolder: RemoteFolder,
+        override val isAutomatic: Boolean = false,
+    ) : SpecialFolder
+
+    data class Outbox(
+        override val remoteFolder: RemoteFolder,
+        override val isAutomatic: Boolean = false,
+    ) : SpecialFolder
+
+    data class Archive(
+        override val remoteFolder: RemoteFolder,
+        override val isAutomatic: Boolean = false,
+    ) : SpecialFolder
+
+    data class Drafts(
+        override val remoteFolder: RemoteFolder,
+        override val isAutomatic: Boolean = false,
+    ) : SpecialFolder
+
+    data class Sent(
+        override val remoteFolder: RemoteFolder,
+        override val isAutomatic: Boolean = false,
+    ) : SpecialFolder
+
+    data class Spam(
+        override val remoteFolder: RemoteFolder,
+        override val isAutomatic: Boolean = false,
+    ) : SpecialFolder
+
+    data class Trash(
+        override val remoteFolder: RemoteFolder,
+        override val isAutomatic: Boolean = false,
+    ) : SpecialFolder
+}
```

Comment:
This is meant to avoid assigning Inbox to Trash or other combinations. But we agreed to not check that at them moment. Nevertheless I would like to keep this, it simplifies any later check for consisten folder mapping taking the type of folder into account

Reply:
We still want to allow the user to assign any folder they want to a special role, no matter what role the server assigned to that folder. We do want to stop them from assigning the same folder to two special roles in K-9 Mail. I don't think this helps with that.

Reply:
I still think this is more a bug than a feature and from the email clients I tested, it is either not possible to configure the folders at all or just assign either a matching type or a regular folder. But anyway this is a discussion for later.